src/main/java/org/folio/rest/impl/PutOrdersByIdHelper.java

```diff
@@ -0,0 +1,103 @@
+package org.folio.rest.impl;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import javax.ws.rs.core.Response;
+
+import org.apache.log4j.Logger;
+import org.folio.orders.rest.exceptions.HttpException;
+import org.folio.rest.jaxrs.model.CompositePurchaseOrder;
+import org.folio.rest.jaxrs.resource.OrdersResource.PutOrdersByIdResponse;
+import org.folio.rest.tools.client.interfaces.HttpClientInterface;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.json.JsonObject;
+import me.escoffier.vertx.completablefuture.VertxCompletableFuture;
+
+public class PutOrdersByIdHelper {
+  private static final Logger logger = Logger.getLogger(DeleteOrdersByIdHelper.class);
+
+  private final HttpClientInterface httpClient;
+  private final Context ctx;
+  private final Handler<AsyncResult<javax.ws.rs.core.Response>> asyncResultHandler;
+  private final Map<String, String> okapiHeaders;
+
+  public PutOrdersByIdHelper(HttpClientInterface httpClient, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context ctx) {
+    Map<String, String> customHeader = new HashMap<>();
+    customHeader.put(HttpHeaders.ACCEPT.toString(), "application/json, text/plain");
+    httpClient.setDefaultHeaders(customHeader);
+
+    this.httpClient = httpClient;
+    this.okapiHeaders = okapiHeaders;
+    this.ctx = ctx;
+    this.asyncResultHandler = asyncResultHandler;
+  }
+  
+  /*
+   * Handle update order by doing Delete and Post
+   */
+  public CompletableFuture<Void> updateOrder(String id, String lang, CompositePurchaseOrder compPO,
+      Context vertxContext) {
+    CompletableFuture<Void> future = new VertxCompletableFuture<>(ctx);
+
+    PostOrdersHelper postHelper = new PostOrdersHelper(httpClient, okapiHeaders, asyncResultHandler, vertxContext);
+    DeleteOrdersByIdHelper delHelper = new DeleteOrdersByIdHelper(httpClient, okapiHeaders, asyncResultHandler,
+        vertxContext);
+
+    delHelper.deleteOrder(id, lang)
+    .thenRun(() -> {
+      try {
+        compPO.setId(id);
+        postHelper.createPOandPOLines(compPO)
```

Comment:
I think it would be better to also add the place holders that Post had for "apply funds" and "update inventory" here, so that we need not do extra work when those are implemented

Reply:
Agree